### DISCUSSION TITLE
Enable plus/minus keys (+/-) to zoom in/out to preset levels on macOS

### DISCUSSION THREAD

**Discussion by gammalogic** [jgraph/drawio#4159] [General]
Using drawio on a Mac laptop, it would be helpful to have a way to zoom in/out to preset levels using the keyboard. Mac laptops generally do not have a numeric keypad, so the existing **Cmd + Numpad** keyboard shortcut to do this is unavailable, and I find the **Alt + Mousewheel** combination somewhat imprecise (these zoom controls seem to operate too quickly and I do not want to zoom in to abitrary zoom levels like 205%).

Ideally (for me) being able to use **Cmd +** and **Cmd -** to zoom in/out to preset levels would be very helpful as I frequently need to zoom in to make sure flowchart elements are aligned correctly while I'm drawing them, then zoom out to see the complete flowchart. AFAIK the **+** and **-** keys to not appear to be mapped to any other functions, so this would be a helpful addition. The **Cmd +** and **Cmd -** key functionality is fairly prevalent in a lot of common macos applications, so it would be logical to add it to drawio.

Thank you!

**Comment by alderg** [comment]
In Chrome on macOS, Cmd + and Cmd - are mapped to Make Text Larger/Smaller.

**Reply by gammalogic** [reply]
Yes you are right, there is a conflict there. I've added some other comments to the discussion about this.

**Comment by gammalogic** [comment]
The **Cmd +** and **Cmd -** controls for zooming in/out are fairly ubiquitous in macOS. Firefox also zooms in/out using these controls, as does Preview, etc. Firefox calls this action "Zoom In" and "Zoom Out" but by default a "Zoom Text Only" option is set on these options so that only the text is made bigger/smaller (however, the effect is to zoom in to the page). My suggestion was made because the default zoom in/out keyboard mappings in drawio for macOS don't necessarily work across all machines as the keys to use this mapping don't exist, therefore it would be helpful to have a mapping that works on these machines and is consistent with other macOS applications.

**Comment by gammalogic** [comment]
I should just add that I'm aware drawio is also used in a browser, so the suggested changes to the keyboard mapping would be **only** for the macOS standalone application (which is what I use). Perhaps there is an alternative set of keys that could be used to keep the keyboard mappings consistent across platforms.

**Comment by datgausaigon** [comment]
Hi all,
Can we write a plugin to temporarily fix this issue?
Plugin allows to change Shortcut Zoom In / Zoom Out.

**Comment by datgausaigon** [comment]
Hi all, 
Please take a look at this related Issue in drawio-desktop for macos

https://github.com/jgraph/drawio-desktop/issues/978

**Comment by alderg** [comment]
You do know that there is pinch to zoom (touchpad) and enter to reset/fit the view?

**Comment by davidjgraph** [comment]
Since the OP is using desktop, I'm marking this as a duplicate of https://github.com/jgraph/drawio-desktop/issues/978.
